pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,438 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transforms for interacting with PyZX, framework for ZX calculus."""
+# pylint: disable=too-many-statements, too-many-branches, too-many-return-statements, too-many-arguments
+
+from collections import OrderedDict
+import numpy as np
+import pennylane as qml
+from pennylane.transforms.op_transforms import op_transform
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+class VertexType:  # pylint: disable=too-few-public-methods
+    """Type of a vertex in the graph."""
+
+    BOUNDARY = 0
+    Z = 1
+    X = 2
+    H_BOX = 3
+
+
+class EdgeType:  # pylint: disable=too-few-public-methods
+    """Type of an edge in the graph."""
+
+    SIMPLE = 1
```

Comment:
does this enum need to start at 1? I see the [`enum.Enum`](https://docs.python.org/3/library/enum.html) type often starts this way, but your other enum starts at 0. 1-indexing always feels funny but I guess my ask is if there's a strong reason or otherwise, if this and VertexType can agree. ty! ps: if you want to avoid that pylint thing you can use the functional syntax from that enum doc.

Reply:
For context here, I could not import `class EdgeType` in the whole file and could not use it in the private functions without adding a try and except. So I copied the two classes from PyZX, therefore the definition was not changed.

Reply:
ahh, I see you copied it directly from [PyZX source](https://github.com/Quantomatic/pyzx/blob/ad83429c1ca26c4852092cbdeb7dd6c136618e1f/pyzx/utils.py#L27-L49). Can you add a comment directly into the code? This would be important as this is likely tied to the version of PyZX that the user is running. The fact that it isn't a PennyLane dependency makes this a bit scarier... but we could perhaps document which PyZX versions we support in our docs?

Reply:
Yes I added the copyright as suggested by you and @josh146. I will add a comment about version in the docstrings!